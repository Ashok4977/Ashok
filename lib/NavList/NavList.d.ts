import { ForwardRefComponent as PolymorphicForwardRefComponent } from '../utils/polymorphic';
import React from 'react';
import { ActionListDividerProps, ActionListLeadingVisualProps, ActionListTrailingVisualProps } from '../ActionList';
import { SxProp } from '../sx';
export declare type NavListProps = {
    children: React.ReactNode;
} & SxProp & React.ComponentProps<'nav'>;
export declare type NavListItemProps = {
    children: React.ReactNode;
    href?: string;
    'aria-current'?: 'page' | 'step' | 'location' | 'date' | 'time' | 'true' | 'false' | boolean;
} & SxProp;
export declare type NavListSubNavProps = {
    children: React.ReactNode;
} & SxProp;
export declare type NavListLeadingVisualProps = ActionListLeadingVisualProps;
export declare type NavListTrailingVisualProps = ActionListTrailingVisualProps;
export declare type NavListDividerProps = ActionListDividerProps;
export declare type NavListGroupProps = {
    children: React.ReactNode;
    title?: string;
} & SxProp;
export declare const NavList: React.ForwardRefExoticComponent<Pick<NavListProps, "key" | "sx" | keyof React.HTMLAttributes<HTMLElement>> & React.RefAttributes<HTMLElement>> & {
    Item: PolymorphicForwardRefComponent<"a", NavListItemProps>;
    SubNav: {
        ({ children, sx: sxProp }: NavListSubNavProps): JSX.Element | null;
        displayName: string;
    };
    LeadingVisual: React.FC<React.PropsWithChildren<SxProp & React.HTMLAttributes<HTMLSpanElement>>>;
    TrailingVisual: React.FC<React.PropsWithChildren<SxProp & React.HTMLAttributes<HTMLSpanElement>>>;
    Divider: React.FC<React.PropsWithChildren<SxProp>>;
    Group: React.VFC<NavListGroupProps>;
};
