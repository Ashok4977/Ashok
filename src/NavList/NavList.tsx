import React from 'react'
import {ActionList} from '../ActionList'

// ----------------------------------------------------------------------------
// NavList

export type NavListProps = {
  'aria-label'?: string
}

const Root = React.forwardRef<HTMLElement, NavListProps>(({children}, ref) => {
  return (
    <nav ref={ref}>
      <ActionList>{children}</ActionList>
    </nav>
  )
})

Root.displayName = 'NavList'

// ----------------------------------------------------------------------------
// NavList.Item

export type NavListItemProps = {
  href?: string
  'aria-current'?: 'page' | 'step' | 'location' | 'date' | 'time' | 'true' | 'false' | boolean
}

const Item = React.forwardRef<HTMLAnchorElement, NavListItemProps>(
  ({href, 'aria-current': ariaCurrent, children}, ref) => {
    return (
      <ActionList.LinkItem ref={ref} href={href} aria-current={ariaCurrent}>
        {children}
      </ActionList.LinkItem>
    )
  }
)

// ----------------------------------------------------------------------------
// Export

export const NavList = Object.assign(Root, {
  Item
})
