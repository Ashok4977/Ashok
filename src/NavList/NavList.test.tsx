import {render} from '@testing-library/react'
import React from 'react'
import {ThemeProvider, SSRProvider} from '..'
import {NavList} from './NavList'

describe('NavList', () => {
  it('renders a simple list', () => {
    const {container} = render(
      <ThemeProvider>
        <SSRProvider>
          <NavList>
            <NavList.Item href="/" aria-current="page">
              Home
            </NavList.Item>
            <NavList.Item href="/about">About</NavList.Item>
            <NavList.Item href="/contact">Contact</NavList.Item>
          </NavList>
        </SSRProvider>
      </ThemeProvider>
    )
    expect(container).toMatchSnapshot()
  })

  it('renders with groups', () => {
    const {container} = render(
      <ThemeProvider>
        <SSRProvider>
          <NavList>
            <NavList.Group title="Overview">
              <NavList.Item href="/getting-started" aria-current="page">
                Getting started
              </NavList.Item>
            </NavList.Group>
            <NavList.Group title="Components">
              <NavList.Item href="/Avatar">Avatar</NavList.Item>
            </NavList.Group>
          </NavList>
        </SSRProvider>
      </ThemeProvider>
    )
    expect(container).toMatchSnapshot()
  })
})

describe('NavList.Item', () => {
  it.todo('passes aria-current prop to the underlying <a> element')

  it.todo('has active styles if aria-current is set')
})

describe('NavList.Item with NavList.SubNav', () => {
  it.todo('renders as a button')

  it.todo('ignores aria-current prop')

  it.todo('shows SubNav by default if SubNav contains the current item')

  it.todo('hides SubNav by default if SubNav does not contain the current item')

  it.todo('toggles visiblility of SubNav when clicked')

  it.todo('has active styles if SubNav contains the current item and is closed')

  it.todo('does not have active styles if SubNav contains the current item and is open')

  it.todo('prints an error if Item contains multiple nested SubNavs')
})
