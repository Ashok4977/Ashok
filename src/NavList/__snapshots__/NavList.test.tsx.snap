// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavList renders a simple list 1`] = `
.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  min-width: 0;
}

.c5 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c0 {
  margin: 0;
  padding-inline-start: 0;
  padding-top: 8px;
  padding-bottom: 8px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding-left: 0;
  padding-right: 0;
  font-size: 14px;
  padding-top: 0;
  padding-bottom: 0;
  line-height: 20px;
  min-height: 5px;
  margin-left: 8px;
  margin-right: 8px;
  border-radius: 6px;
  -webkit-transition: background 33.333ms linear;
  transition: background 33.333ms linear;
  color: #24292f;
  cursor: pointer;
}

.c2[aria-disabled] {
  cursor: not-allowed;
}

.c2 [data-component="ActionList.Item--DividerContainer"] {
  position: relative;
}

.c2 [data-component="ActionList.Item--DividerContainer"]::before {
  content: " ";
  display: block;
  position: absolute;
  width: 100%;
  top: -7px;
  border: 0 solid;
  border-top-width: 0;
  border-color: var(--divider-color,transparent);
}

.c2:not(:first-of-type) {
  --divider-color: rgba(208,215,222,0.48);
}

[data-component="ActionList.Divider"] + .c2 {
  --divider-color: transparent !important;
}

.c2:hover:not([aria-disabled]),
.c2:focus:not([aria-disabled]),
.c2[data-focus-visible-added]:not([aria-disabled]) {
  --divider-color: transparent;
}

.c2:hover:not([aria-disabled]) + .c1,
.c2:focus:not([aria-disabled]) + .c2,
.c2[data-focus-visible-added] + li {
  --divider-color: transparent;
}

.c3 {
  color: #0969da;
  -webkit-text-decoration: none;
  text-decoration: none;
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 6px;
  padding-bottom: 6px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  border-radius: 6px;
  color: inherit;
  position: relative;
}

.c3:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c3:is(button) {
  display: inline-block;
  padding: 0;
  font-size: inherit;
  white-space: nowrap;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  background-color: transparent;
  border: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
}

.c3:hover {
  color: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c3[aria-current] {
  font-weight: 600;
  background-color: rgba(208,215,222,0.24);
}

.c3[aria-current]::after {
  position: absolute;
  top: calc(50% - 12px);
  left: -8px;
  width: 4px;
  height: 24px;
  content: "";
  background-color: #0969da;
  border-radius: 6px;
}

@media (hover:hover) and (pointer:fine) {
  .c2:hover:not([aria-disabled]) {
    background-color: rgba(208,215,222,0.32);
    color: #24292f;
  }

  .c2:focus:not([data-focus-visible-added]) {
    background-color: rgba(208,215,222,0.24);
    color: #24292f;
    outline: none;
  }

  .c2[data-focus-visible-added] {
    outline: none;
    border: 2 solid;
    box-shadow: 0 0 0 2px #0969da;
  }

  .c2:active:not([aria-disabled]) {
    background-color: rgba(208,215,222,0.48);
    color: #24292f;
  }
}

@media (forced-colors:active) {
  .c2:focus {
    outline: solid 1px transparent !important;
  }
}

<div>
  <nav>
    <ul
      class="c0"
    >
      <li
        aria-labelledby="react-aria-1 "
        class="c1 c2"
      >
        <a
          aria-current="page"
          class="c3"
          href="/"
        >
          <div
            class="c4"
            data-component="ActionList.Item--DividerContainer"
          >
            <span
              class="c5"
              id="react-aria-1"
            >
              Home
            </span>
          </div>
        </a>
      </li>
      <li
        aria-labelledby="react-aria-4 "
        class="c1 c2"
      >
        <a
          class="c3"
          href="/about"
        >
          <div
            class="c4"
            data-component="ActionList.Item--DividerContainer"
          >
            <span
              class="c5"
              id="react-aria-4"
            >
              About
            </span>
          </div>
        </a>
      </li>
      <li
        aria-labelledby="react-aria-7 "
        class="c1 c2"
      >
        <a
          class="c3"
          href="/contact"
        >
          <div
            class="c4"
            data-component="ActionList.Item--DividerContainer"
          >
            <span
              class="c5"
              id="react-aria-7"
            >
              Contact
            </span>
          </div>
        </a>
      </li>
    </ul>
  </nav>
</div>
`;

exports[`NavList renders with groups 1`] = `
.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  min-width: 0;
}

.c8 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c1 {
  list-style: none;
}

.c1:not(:first-child) {
  margin-top: 8px;
}

.c2 {
  padding-top: 6px;
  padding-bottom: 6px;
  padding-left: 16px;
  padding-right: 16px;
  font-size: 12px;
  font-weight: 600;
  color: #57606a;
}

.c3 {
  padding-inline-start: 0;
}

.c0 {
  margin: 0;
  padding-inline-start: 0;
  padding-top: 8px;
  padding-bottom: 8px;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding-left: 0;
  padding-right: 0;
  font-size: 14px;
  padding-top: 0;
  padding-bottom: 0;
  line-height: 20px;
  min-height: 5px;
  margin-left: 8px;
  margin-right: 8px;
  border-radius: 6px;
  -webkit-transition: background 33.333ms linear;
  transition: background 33.333ms linear;
  color: #24292f;
  cursor: pointer;
}

.c5[aria-disabled] {
  cursor: not-allowed;
}

.c5 [data-component="ActionList.Item--DividerContainer"] {
  position: relative;
}

.c5 [data-component="ActionList.Item--DividerContainer"]::before {
  content: " ";
  display: block;
  position: absolute;
  width: 100%;
  top: -7px;
  border: 0 solid;
  border-top-width: 0;
  border-color: var(--divider-color,transparent);
}

.c5:not(:first-of-type) {
  --divider-color: rgba(208,215,222,0.48);
}

[data-component="ActionList.Divider"] + .c5 {
  --divider-color: transparent !important;
}

.c5:hover:not([aria-disabled]),
.c5:focus:not([aria-disabled]),
.c5[data-focus-visible-added]:not([aria-disabled]) {
  --divider-color: transparent;
}

.c5:hover:not([aria-disabled]) + .c4,
.c5:focus:not([aria-disabled]) + .c5,
.c5[data-focus-visible-added] + li {
  --divider-color: transparent;
}

.c6 {
  color: #0969da;
  -webkit-text-decoration: none;
  text-decoration: none;
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 6px;
  padding-bottom: 6px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  border-radius: 6px;
  color: inherit;
  position: relative;
}

.c6:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c6:is(button) {
  display: inline-block;
  padding: 0;
  font-size: inherit;
  white-space: nowrap;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  background-color: transparent;
  border: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
}

.c6:hover {
  color: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c6[aria-current] {
  font-weight: 600;
  background-color: rgba(208,215,222,0.24);
}

.c6[aria-current]::after {
  position: absolute;
  top: calc(50% - 12px);
  left: -8px;
  width: 4px;
  height: 24px;
  content: "";
  background-color: #0969da;
  border-radius: 6px;
}

@media (hover:hover) and (pointer:fine) {
  .c5:hover:not([aria-disabled]) {
    background-color: rgba(208,215,222,0.32);
    color: #24292f;
  }

  .c5:focus:not([data-focus-visible-added]) {
    background-color: rgba(208,215,222,0.24);
    color: #24292f;
    outline: none;
  }

  .c5[data-focus-visible-added] {
    outline: none;
    border: 2 solid;
    box-shadow: 0 0 0 2px #0969da;
  }

  .c5:active:not([aria-disabled]) {
    background-color: rgba(208,215,222,0.48);
    color: #24292f;
  }
}

@media (forced-colors:active) {
  .c5:focus {
    outline: solid 1px transparent !important;
  }
}

<div>
  <nav>
    <ul
      class="c0"
    >
      <li
        class="c1"
      >
        <div
          aria-hidden="true"
          class="c2"
          role="presentation"
        >
          <span
            id="react-aria-1"
          >
            Overview
          </span>
        </div>
        <ul
          aria-labelledby="react-aria-1"
          class="c3"
        >
          <li
            aria-labelledby="react-aria-2 "
            class="c4 c5"
          >
            <a
              aria-current="page"
              class="c6"
              href="/getting-started"
            >
              <div
                class="c7"
                data-component="ActionList.Item--DividerContainer"
              >
                <span
                  class="c8"
                  id="react-aria-2"
                >
                  Getting started
                </span>
              </div>
            </a>
          </li>
        </ul>
      </li>
      <li
        class="c1"
      >
        <div
          aria-hidden="true"
          class="c2"
          role="presentation"
        >
          <span
            id="react-aria-5"
          >
            Components
          </span>
        </div>
        <ul
          aria-labelledby="react-aria-5"
          class="c3"
        >
          <li
            aria-labelledby="react-aria-6 "
            class="c4 c5"
          >
            <a
              class="c6"
              href="/Avatar"
            >
              <div
                class="c7"
                data-component="ActionList.Item--DividerContainer"
              >
                <span
                  class="c8"
                  id="react-aria-6"
                >
                  Avatar
                </span>
              </div>
            </a>
          </li>
        </ul>
      </li>
    </ul>
  </nav>
</div>
`;
